src/components/chips/js/chipController.js

```diff
@@ -0,0 +1,187 @@
+angular
+  .module('material.components.chips')
+  .controller('MdChipCtrl', MdChipCtrl);
+
+/**
+ * Controller for the MdChip component. Responsible for handling keyboard
+ * events and editting the chip if needed.
+ *
+ * @param $scope
+ * @param $element
+ * @param $mdConstant
+ * @param $timeout
+ * @constructor
+ */
+function MdChipCtrl ($scope, $element, $mdConstant, $timeout) {
+  /**
+   * @type {$scope}
+   */
+  this.$scope = $scope;
+
+  /**
+   * @type {$element}
+   */
+  this.$element = $element;
+
+  /**
+   * @type {$mdConstant}
+   */
+  this.$mdConstant = $mdConstant;
+
+  /**
+   * @type {$timeout}
+   */
+  this.$timeout = $timeout;
+
+  /**
+   * @type {boolean}
+   */
+  this.isEditting = false;
+
+  /**
+   * @type {MdChipsCtrl}
+   */
+  this.parentController = undefined;
+
+  /**
+   * @type {boolean}
+   */
+  this.enableChipEdit = false;
+}
+
+
+/**
+ * @param {MdChipsCtrl} controller
+ */
+MdChipCtrl.prototype.init = function(controller) {
+  this.parentController = controller;
+  this.enableChipEdit = this.parentController.enableChipEdit;
+
+  if (this.enableChipEdit) {
+    this.$element.on('keydown', this.chipKeyDown.bind(this));
+    this.$element.on('mousedown', this.chipMouseDown.bind(this));
+    this.getChipContent().addClass('_md-chip-content-edit-is-enabled');
+  }
+};
+
+
+/**
+ * @return {Object}
+ */
+MdChipCtrl.prototype.getChipContent = function() {
+  var chipContents = this.$element[0].getElementsByClassName('_md-chip-content');
+  return angular.element(chipContents[0]);
+};
+
+
+/**
+ * @return {Object}
+ */
+MdChipCtrl.prototype.getChipContent = function() {
+  var chipContents = this.$element[0].getElementsByClassName('_md-chip-content');
+  return angular.element(chipContents[0]);
+};
+
+
+/**
+ * @return {Object}
+ */
+MdChipCtrl.prototype.getContentElement = function() {
+  return angular.element(this.getChipContent().children()[0]);
+};
+
+
+/**
+ * @return {number}
+ */
+MdChipCtrl.prototype.getChipIndex = function() {
+  return parseInt(this.$element.attr('index'));
+};
+
+
+/**
+ * Presents an input element to edit the contents of the chip.
+ */
+MdChipCtrl.prototype.goOutOfEditMode = function() {
+  this.isEditting = false;
+  this.$element.removeClass('_md-chip-editing');
+  this.getChipContent()[0].contentEditable = 'false';
+  var chipIndex = this.getChipIndex();
+
+  var content = this.getContentElement().text();
+  if (content) {
+    this.parentController.updateChipContents(
+        chipIndex,
+        this.getContentElement().text()
+    );
```

Comment:
Good catch! I have added this code to my changelist.

On Wed, Mar 2, 2016 at 12:21 PM, Paul Gschwendtner <notifications@github.com
> wrote:

> In src/components/chips/js/chipController.js
> <https://github.com/angular/material/pull/7379#discussion_r54784600>:
>
> > +
> > +/**
> > + * Presents an input element to edit the contents of the chip.
> > + */
> > +MdChipCtrl.prototype.goOutOfEditMode = function() {
> > +  this.isEditting = false;
> > +  this.$element.removeClass('_md-chip-editing');
> > +  this.getChipContent()[0].contentEditable = 'false';
> > +  var chipIndex = this.getChipIndex();
> > +
> > +  var content = this.getContentElement().text();
> > +  if (content) {
> > +    this.parentController.updateChipContents(
> > +        chipIndex,
> > +        this.getContentElement().text()
> > +    );
>
> Indeed, that seems to be a create a new problem, when clicking on another
> chip while editing.
>
> I will tell you at first the steps to reproduce.
>
>    - Select Input
>    - Backspace to Focus Chip (ready to delete)
>    - Press Space
>    - Write something new
>    - Press enter to update chip
>    - And then press space again to edit again
>    - But here it *breaks* - because body will be selected
>
> I've just find a way to fix that issue, without introducing a new problem
> (as you mentioned, when clicking on another chip)
>
>     this.$mdUtil.nextTick(function() {
>       if (this.parentController.selectedChip === chipIndex) {
>         this.parentController.focusChip(chipIndex);
>       }
>     }.bind(this));
>
> It's important that the check also runs after the tick.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/7379/files#r54784600>.
>